flocker/control/test/test_protocol.py

```diff
@@ -574,6 +575,55 @@ def test_nodestate_notifies_all_connected(self):
             ),
         )
 
+    def test_nodestate_coalesces_multiple_quick(self):
+        """
+        Multiple ``NodeStateCommands`` are coalesced into a single state update
+        broadcast to all the nodes.
+        """
+        self.control_amp_service.configuration_service.save(TEST_DEPLOYMENT)
+
+        agents = [FakeAgent(), FakeAgent()]
+        clients = list(AgentAMP(Clock(), agent) for agent in agents)
+        servers = list(LoopbackAMPClient(client.locator) for client in clients)
+
+        for server in servers:
+            self.control_amp_service.connected(server)
+
+        initial_update_counts = list(
+            agent.cluster_updated_count for agent in agents)
+
+        for i in xrange(10):
+            new_state = NODE_STATE.set(
+                'applications',
+                NODE_STATE.applications.add(
+                    Application(name=('app-%d' % i),
+                                image=DockerImage.from_string('image-%d' % i))
+                )
+            )
```

Comment:
* Probably not realistic. (state updates should actually come from different nodes)
* Close enough approximation (If I take out the code I wrote, this test fails, yay TDD)
* I actually had to remove the DelayedAmpClient code from this test so that this was not the case. Without the delayed update code the clients respond immediately, so cluster_updated_count increments 10 times in this loop, just as desired.